### DISCUSSION TITLE
Registering Multiple Extension Files

### DISCUSSION THREAD

**Discussion by cemalgnlts** [CodinGame/monaco-vscode-api#295] [Q&A] [answered]
Hi,

I created an extension for icons and this is my main file:
```ts
import { registerExtension, ExtensionHostKind } from "vscode/extensions";
import { IExtensionManifest } from "vscode/vscode/vs/platform/extensions/common/extensions";

const extension: IExtensionManifest = {
	name: "vsc-material-theme",
	publisher: "Philipp Kief",
	version: "4.32.0",
	engines: {
		vscode: "*"
	},
	contributes: {
		iconThemes: [
			{
				/* @ts-ignore */
				id: "material-icon-theme",
				label: "Material Icon Theme",
				path: "./material-icons.json"
			}
		]
	}
};

const { registerFileUrl } = registerExtension(extension, ExtensionHostKind.LocalProcess);

registerFileUrl(
	"/material-icons.json",
	new URL("./material-icons.json", import.meta.url).toString()
);

registerFileUrl(
    "./icons/folder-root-open.svg",
    new URL("./icons/folder-root-open.svg", import.meta.url).toString()
);

registerFileUrl(
    "./icons/javascript.svg",
    new URL("./icons/javascript.svg", import.meta.url).toString()
);

registerFileUrl(
    "./icons/typescript.svg",
    new URL("./icons/typescript.svg", import.meta.url).toString()
);
```

If I don't specify a path for each file individually, it gives an error. Since the icon file is json, of course I can read the file and add it in the loop, but I wanted to ask: Can I use it without specifying all the required files individually? Maybe something like: `registerRootFolder("./icons", new URL("./icons", import.meta.url).toString())`;

**Comment by CGNonofr** [comment, marked as answer]
You'll need your bundler to generate an url (probably hashed) for each asset.
 
You'll also need to call registerFileUrl for each file.

What you can do is using a plugins like https://www.npmjs.com/package/rollup-plugin-glob-import maybe and looping by hands on the result

You can also use the provided rollup plugins

**Reply by cemalgnlts** [reply]
Cool, thanks.

**Reply by cemalgnlts** [reply]
While searching for a glob import package in vite I found that it is already supported: https://vitejs.dev/guide/features#glob-import

```js
const icons = import.meta.glob("./icons/*.svg");

for (const path of Object.keys(icons)) {
  registerFileUrl(path, new URL(path, import.meta.url).toString())
}
```
